{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7a4cf90e-b607-4ef3-a457-916879e24e2a",
   "metadata": {},
   "source": [
    "# ABIO558 Assignment Check-In\n",
    "\n",
    "### Project Goals:\n",
    "1. Write markdown text describing the question and outlining your plan for analysis\n",
    "2. Insert a snippet of code to load data\n",
    "3. print() or visualization a command to view the data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a9e26d0-912d-49d5-a2da-756465c38edc",
   "metadata": {},
   "source": [
    "## Goal 1: Outline your plan for analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8a45b2e-30f9-4b6e-8d97-cfa7084ffd8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Plan for Analysis: Analyze MKD8 WT vs espM KO RNASeq data to determine gene expression differences\n",
    "# --> conditions are WT and deletion of the espM gene. Looking for significant changes\n",
    "# in expression of gene in the whiB6 regulon (and/or anywhere else in the genome)\n",
    "\n",
    "# Outline:\n",
    "\n",
    "# Pull out raw count data for each of the 4 samples\n",
    "\n",
    "# Normalize and calculate p-values\n",
    "\n",
    "# Bonferroni/Multiple testing correction\n",
    "# Note: this gives corrected p-values to account for false significance\n",
    "\n",
    "# Find ratio of KO/WT\n",
    "\n",
    "# Make a log2 expression of the ratio\n",
    "\n",
    "# logfold2 and then transform that into log2\n",
    "\n",
    "# generate Volcano and MA plots\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "281f4fa6-427a-4116-82d4-eb37d6e45ae6",
   "metadata": {},
   "source": [
    "## Goal 2: Make Code to Load Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb5b04e9-e032-46b7-82bc-b8d01b9f6538",
   "metadata": {},
   "source": [
    "### Step 1: Pull out the correct data columns\n",
    "\n",
    "1) Import libraries\n",
    "2) Find file path - check that it's correct\n",
    "3) Load data with pandas\n",
    "4) Pull out raw count columns for each of the 4 samples (note: rename columns - Biorep 1 = OLD, Biorep 2 = NEW)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9e407c8a-f0b0-4b6b-8e4b-62c99af14591",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 1. Import Libraries\n",
    "\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import os\n",
    "import scanpy\n",
    "import anndata\n",
    "import seaborn as sns\n",
    "import skimage\n",
    "from skimage.io import imread\n",
    "from scipy import stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "2c97d65c-5b38-42f7-b858-c1cc0d8f7a02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 2. Find file path\n",
    "\n",
    "data_directory = 'C:/Users/denis/Downloads/'\n",
    "filepath_mkd8 = os.path.join(data_directory, 'MKD8_KO_Rockhopper_data.csv')\n",
    "os.path.exists(filepath_mkd8)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "baf3e17e-235f-40f3-b5ff-ddf9d66fe054",
   "metadata": {},
   "source": [
    "### Goal 3: Visualize a command"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "42581586-9307-4254-bc6e-ae40c28b3155",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Transcription Start</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>2072.0</th>\n",
       "      <th>3303.0</th>\n",
       "      <th>4225.0</th>\n",
       "      <th>NaN</th>\n",
       "      <th>6201.0</th>\n",
       "      <th>8234.0</th>\n",
       "      <th>8250.0</th>\n",
       "      <th>10839.0</th>\n",
       "      <th>...</th>\n",
       "      <th>6979941.0</th>\n",
       "      <th>6982120.0</th>\n",
       "      <th>6983355.0</th>\n",
       "      <th>6984329.0</th>\n",
       "      <th>6985074.0</th>\n",
       "      <th>6985091.0</th>\n",
       "      <th>6985687.0</th>\n",
       "      <th>6987027.0</th>\n",
       "      <th>6987520.0</th>\n",
       "      <th>6987853.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Translation Start</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1454.0</td>\n",
       "      <td>2109.0</td>\n",
       "      <td>3331.0</td>\n",
       "      <td>4234.0</td>\n",
       "      <td>5385.0</td>\n",
       "      <td>6201.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8258.0</td>\n",
       "      <td>10839.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6979984.0</td>\n",
       "      <td>6982002.0</td>\n",
       "      <td>6983296.0</td>\n",
       "      <td>6984327.0</td>\n",
       "      <td>6985067.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6985652.0</td>\n",
       "      <td>6986791.0</td>\n",
       "      <td>6987520.0</td>\n",
       "      <td>6987695.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Translation Stop</th>\n",
       "      <td>1515.0</td>\n",
       "      <td>2071.0</td>\n",
       "      <td>3302.0</td>\n",
       "      <td>4224.0</td>\n",
       "      <td>5388.0</td>\n",
       "      <td>5969.0</td>\n",
       "      <td>8228.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10786.0</td>\n",
       "      <td>11621.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6981204.0</td>\n",
       "      <td>6981250.0</td>\n",
       "      <td>6982241.0</td>\n",
       "      <td>6983356.0</td>\n",
       "      <td>6984330.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6985101.0</td>\n",
       "      <td>6985688.0</td>\n",
       "      <td>6987128.0</td>\n",
       "      <td>6987552.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Transcription Stop</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2071.0</td>\n",
       "      <td>3302.0</td>\n",
       "      <td>4224.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6199.0</td>\n",
       "      <td>8233.0</td>\n",
       "      <td>8249.0</td>\n",
       "      <td>10838.0</td>\n",
       "      <td>11681.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6980814.0</td>\n",
       "      <td>6982126.0</td>\n",
       "      <td>6983356.0</td>\n",
       "      <td>6984330.0</td>\n",
       "      <td>6985075.0</td>\n",
       "      <td>6985092.0</td>\n",
       "      <td>6985688.0</td>\n",
       "      <td>6987033.0</td>\n",
       "      <td>6987521.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strand</th>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>+</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <td>dnaA</td>\n",
       "      <td>-</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>-</td>\n",
       "      <td>recF</td>\n",
       "      <td>-</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>-</td>\n",
       "      <td>gyrA</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>parB</td>\n",
       "      <td>-</td>\n",
       "      <td>gidB</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>oxaA</td>\n",
       "      <td>rnpA</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 6891 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Transcription Start NaN       NaN       2072.0    3303.0    4225.0     \\\n",
       "Translation Start         1.0    1454.0    2109.0    3331.0    4234.0   \n",
       "Translation Stop       1515.0    2071.0    3302.0    4224.0    5388.0   \n",
       "Transcription Stop        NaN    2071.0    3302.0    4224.0       NaN   \n",
       "Strand                      +         +         +         +         +   \n",
       "Name                     dnaA         -      dnaN         -      recF   \n",
       "\n",
       "Transcription Start NaN       6201.0    8234.0    8250.0    10839.0    ...  \\\n",
       "Translation Start      5385.0    6201.0       NaN    8258.0   10839.0  ...   \n",
       "Translation Stop       5969.0    8228.0       NaN   10786.0   11621.0  ...   \n",
       "Transcription Stop     6199.0    8233.0    8249.0   10838.0   11681.0  ...   \n",
       "Strand                      +         +         +         +         +  ...   \n",
       "Name                        -      gyrB         -      gyrA         -  ...   \n",
       "\n",
       "Transcription Start  6979941.0  6982120.0  6983355.0  6984329.0  6985074.0  \\\n",
       "Translation Start    6979984.0  6982002.0  6983296.0  6984327.0  6985067.0   \n",
       "Translation Stop     6981204.0  6981250.0  6982241.0  6983356.0  6984330.0   \n",
       "Transcription Stop         NaN  6980814.0  6982126.0  6983356.0  6984330.0   \n",
       "Strand                       +          -          -          -          -   \n",
       "Name                         -          -       parB          -       gidB   \n",
       "\n",
       "Transcription Start  6985091.0  6985687.0  6987027.0  6987520.0  6987853.0  \n",
       "Translation Start          NaN  6985652.0  6986791.0  6987520.0  6987695.0  \n",
       "Translation Stop           NaN  6985101.0  6985688.0  6987128.0  6987552.0  \n",
       "Transcription Stop   6985075.0  6985092.0  6985688.0  6987033.0  6987521.0  \n",
       "Strand                       -          -          -          -          -  \n",
       "Name                         -          -       oxaA       rnpA          -  \n",
       "\n",
       "[5 rows x 6891 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 3. Load data with pandas\n",
    "\n",
    "reads_df = pd.read_csv(filepath, index_col=0).T\n",
    "reads_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "24e4eacd-63c5-4162-bbf2-10cd3c138e8d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Raw Counts MKD8-oldmycospike</th>\n",
       "      <th>Raw Counts MKD8-NEWmycospike</th>\n",
       "      <th>Raw Counts KOespM-OLDmycospike</th>\n",
       "      <th>Raw Counts KOespM-NEWmycospike</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4093.0</td>\n",
       "      <td>3700.0</td>\n",
       "      <td>1442.0</td>\n",
       "      <td>1604.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1969.0</td>\n",
       "      <td>1744.0</td>\n",
       "      <td>749.0</td>\n",
       "      <td>650.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>13295.0</td>\n",
       "      <td>9364.0</td>\n",
       "      <td>4701.0</td>\n",
       "      <td>4894.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8056.0</td>\n",
       "      <td>5592.0</td>\n",
       "      <td>4026.0</td>\n",
       "      <td>3617.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3001.0</td>\n",
       "      <td>1928.0</td>\n",
       "      <td>1513.0</td>\n",
       "      <td>1542.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Raw Counts MKD8-oldmycospike  Raw Counts MKD8-NEWmycospike  \\\n",
       "0                        4093.0                        3700.0   \n",
       "1                        1969.0                        1744.0   \n",
       "2                       13295.0                        9364.0   \n",
       "3                        8056.0                        5592.0   \n",
       "4                        3001.0                        1928.0   \n",
       "\n",
       "   Raw Counts KOespM-OLDmycospike  Raw Counts KOespM-NEWmycospike  \n",
       "0                          1442.0                          1604.0  \n",
       "1                           749.0                           650.0  \n",
       "2                          4701.0                          4894.0  \n",
       "3                          4026.0                          3617.0  \n",
       "4                          1513.0                          1542.0  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 4. extract the raw count columns\n",
    "\n",
    "reads_df_2 = pd.read_csv(filepath_mkd8, usecols=['Raw Counts MKD8-oldmycospike', 'Raw Counts MKD8-NEWmycospike', 'Raw Counts KOespM-OLDmycospike', 'Raw Counts KOespM-NEWmycospike'])\n",
    "reads_df_2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18742a71-3b7f-4abf-bd97-5e7a195365ac",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:base] *",
   "language": "python",
   "name": "conda-base-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
